api/python/cell_census/tests/test_acceptance.py

```diff
@@ -0,0 +1,115 @@
+
+Several of them will not run to completion except on VERY large hosts.
+
+Intended use:  periodically do a manual run, including the expensive tests, on an
+appropriately large host.
+
+See README.md for historical data.
+"""
+from typing import Iterator, Optional
+
+import pyarrow as pa
+import pytest
+import tiledb
+import tiledbsoma as soma
+
+import cell_census
+
+
+@pytest.mark.live_corpus
+@pytest.mark.parametrize("organism", ["homo_sapiens", "mus_musculus"])
+def test_load_axes(organism: str) -> None:
+    """Verify axes can be loaded into a Pandas DataFrame"""
+    census = cell_census.open_soma(census_version="latest")
+
+    # use subset of columns for speed
+    obs_df = (
+        census["census_data"][organism]
+        .obs.read(column_names=["soma_joinid", "cell_type", "tissue"])
+        .concat()
+        .to_pandas()
+    )
+    assert len(obs_df)
+    del obs_df
+
+    var_df = census["census_data"][organism].ms["RNA"].var.read().concat().to_pandas()
+    assert len(var_df)
+    del var_df
+
+
+def table_iter_is_ok(tbl_iter: Iterator[pa.Table], stop_after: Optional[int] = 2) -> bool:
+    """
+    Utility that verifies that the value is an iterator of pa.Table.
+
+    Will only call __next__ as many times as the `stop_after` param specifies,
+    or will read until end of iteration of it is None.
+    """
+    assert isinstance(tbl_iter, Iterator)
+    for n, tbl in enumerate(tbl_iter):
+        # keep things speedy by quitting early if stop_after specified
+        if stop_after is not None and n > stop_after:
+            break
+        assert isinstance(tbl, pa.Table)
+        assert len(tbl)
+
+    return True
+
+
+@pytest.mark.live_corpus
+@pytest.mark.parametrize("organism", ["homo_sapiens", "mus_musculus"])
+def test_incremental_read(organism: str) -> None:
+    """Verify that obs, var and X[raw] can be read incrementally, i.e., in chunks"""
+
+    # open census with a small (default) TileDB buffer size, which reduces
+    # memory use, and makes it feasible to run in a GHA.
+    version = cell_census.get_census_version_description("latest")
+    s3_region = version["soma"].get("s3_region")
+    context = soma.options.SOMATileDBContext(tiledb_ctx=tiledb.Ctx({"vfs.s3.region": s3_region}))
+
+    with cell_census.open_soma(census_version="latest", context=context) as census:
+        assert table_iter_is_ok(census["census_data"][organism].obs.read(column_names=["soma_joinid", "tissue"]))
+        assert table_iter_is_ok(
+            census["census_data"][organism].ms["RNA"].var.read(column_names=["soma_joinid", "feature_id"])
+        )
+        assert table_iter_is_ok(census["census_data"][organism].ms["RNA"].X["raw"].read().tables())
+
+
+@pytest.mark.live_corpus
+@pytest.mark.parametrize("organism", ["homo_sapiens", "mus_musculus"])
+@pytest.mark.parametrize(
+    "obs_value_filter", ["tissue=='aorta'", pytest.param("tissue=='brain'", marks=pytest.mark.expensive)]
```

Comment:
pytest is great; TIL: param values can add marks

Reply:
always pays to RT*M :-)